src/openms/include/OpenMS/QC/DBSuitability.h

```diff
@@ -203,7 +269,128 @@ namespace OpenMS
+    *
+    * @param fasta_data         fasta of which the subsampling should be done
+    * @param subsampling_rate   subsampling rate to be used [0,1]
+    * @returns                  fasta entries with total number of AA = original number of AA * subsampling_rate
+    * @throws                   IllegalArgument if subsampling rate is not between 0 and 1
+    */
+    std::vector<FASTAFile::FASTAEntry> getSubsampledFasta_(std::vector<FASTAFile::FASTAEntry> fasta_data, double subsampling_rate) const;
+
+    /**
+    * @brief Calculates all suitability data from a combined deNovo+database search
+    *
+    * Counts top database and top deNovo hits.
+    *
+    * Calculates a decoy score cut-off to compare high scoring deNovo hits with lower scoring database hits.
+    * If the score difference is smaller than the cut-off the database hit is counted and the deNovo hit ignored.
+    *
+    * Suitability is calculated: # database hits / # all hits
+    *
+    * @param pep_ids    peptide identifications coming from the combined search
+    * @param data       SuitabilityData object where the result should be written into
+    * @throws           MissingInformation if no target/decoy annotation is found on @p pep_ids
+    * @throws           MissingInformation if no xcorr is found,
+    *                   this happends when another adapter than CometAdapter was used
+    */
+    void calculateSuitability_(std::vector<PeptideIdentification> pep_ids, SuitabilityData& data) const;
+
+    /**
+    * @brief Calculates and appends decoys to a given vector of FASTAEntry
+    *
+    * Each sequence is digested with Trypsin. The resulting peptides are reversed and appended to one another.
+    * This results in the decoy sequences.
+    * The idetifier is given a 'DECOY_' prefix.
+    *
+    * @param fasta     reference to fasta vector where the decoys are needed
+    */
+    void appendDecoys_(std::vector<FASTAFile::FASTAEntry>& fasta) const;
+
+    /**
+    * @brief Returns the cross correlation score normalized by MW (if existing), else if the 'force' flag is set the current main score is returned
+    *
+    * @param pep_hit    PeptideHit of which the score is needed
+    * @returns          cross correlation score normalized by MW or current score
+    * @throws           MissingInformation if no xcorr is found and 'force' flag isn't set
+    */
+    double getRightScore_(const PeptideHit& pep_hit) const;
+
+    /**
+    * @brief Returns a SuitabilityData object containing the data if re-ranking didn't happen
+    *
+    * Cases that are re-ranked are already counted. To get the 'no re-ranking' data these cases need to be
+    * subtracted from the number of top database hits and added to the number of top deNovo hits.
+    *
+    * @param data    actual suitability data
+    * @returns       simulated suitability data where re-ranking didn't happen
+    */
+    SuitabilityData simulateNoReRanking_(const SuitabilityData& data) const;
```

Comment:
maybe make this a member of `SuitabilityData`? Just to be closer to the data model?